doc(tutorials): adapt tutorial description to fit new structure/basis (#885)
eclipse-tractusx/eclipse-tractusx.github.io: docs/tutorials/e2e/boost/dtrTutorial.md

Reviewer: Iam not familiar with the structure, but i think the following table is missing: 

title | sidebar_position
-- | --
<Name of component> | <order>

Author: It's not mandatory. If you add a title here, then you have to delete your first `# Header`. Side position would be good, but right now, it's rendered correctly.

Not all files include this table :)

Author: @tunacicek added the table :) Thx

---
title: DTR Tutorial
sidebar_position: 5
---

## Tutorial Goal

Alongside the connector, the Digital Twin Registry (DTR) is another central component in a data ecosystem when it comes to exchanging digital twins. Anyone who provides digital twins in Catena-X must register them in a DTR so that they can be found by other parties.

This tutorial focuses on the working with the DTR. You will learn what the DTR is, how to deploy it and go through a complete use case scenario by depositing a DTR in the EDC, registering a digital twin in the DTR, and showing how external parties can find and consume the digital twin.

## Introduction

> Catena-X uses the Asset Administration Shell (AAS) to represent digital twins. Such a digital twin is basically a shell with an ID in UUID-format that makes it uniquely identifiable. Also it contains IDs ("specificAssetIds") that connect the shell with the original asset it represents (e.g. the "manufacturerPartId"). The shell also contains SubmodelDescriptors which reference to submodels that contain the actual data of specific aspects of the asset.

Reviewer 2: ```suggestion
Catena-X uses the Asset Administration Shell (AAS) to represent digital twins. Such a digital twin is basically a shell with an ID (usually a UUID) that makes it uniquely identifiable. Also it contains IDs ("specificAssetIds") that connect the shell with the original asset it represents (e.g. the "manufacturerPartId"). The shell also contains SubmodelDescriptors which reference to submodels that contain the actual data of specific aspects of the asset.
```
as per spec, aasIds don't have to be UUIDs even though it's a convention

The steps of the tutorial are as follows

- Bob registers a DTR at his EDC
- Bob registers a Digital Twin at his DTR
- Alice finds and consumes the Digital Twin

This tutorial will guide you through the steps outlined above. For a deeper understanding, more detailed documentation will be linked.

### High Level Architecture

## Exchange Digital Twins

### Register a DTR in your EDC

#### Create the DTR asset

To ensure that Bob's DTR becomes visible for Alice and to start the data exchange between them, Bob has to create a data asset.

Action (Bob): Create a data asset using the following command:

(note: that the "asset:prop:type" is standardized with "data.core.digitalTwinRegistry" for the Digital Twin Registry.)

```curl --location 'http://dataprovider-controlplane.tx.test/management/v3/assets' \
--header 'Content-Type: application/json' \
--header 'X-Api-Key: TEST2' \
--data-raw '{
   "@context":{
      "edc":"https://w3id.org/edc/v0.0.1/ns/",
      "cx-common":"https://w3id.org/catenax/ontology/common#",
      "cx-taxo":"https://w3id.org/catenax/taxonomy#",

>       "dct":"https://purl.org/dc/terms/"

Reviewer 2: ```suggestion
      "dct":"http://purl.org/dc/terms/"
```
This will come up multiple times, there used to be an error in a spec somewhere that spread. For reference, check https://www.dublincore.org/specifications/dublin-core/dcmi-terms/#type

Author: @tunacicek and I adapted the URLs together

   },
   "@id":"{{ASSET_ID}}",
   "properties":{
      "dct:type":{
         "@id":"cx-taxo:DigitalTwinRegistry"
      }
   },
   "dataAddress":{
      "@type":"DataAddress",
      "type":"HttpData",
      "baseUrl":"{{BACKEND_SERVICE}}",
      "proxyPath":"true",
      "proxyBody":"true",

>       "proxyMethod":"true",

Reviewer 2: ```suggestion
      "proxyMethod":"false",
```
this is borderline dangerous, see below

      "proxyQueryParams":"true",
      "oauth2:clientId":"satest02",
      "oauth2:clientSecretKey":"{{REGISTRY_CLIENT_SECRET_KEY}}",
      "oauth2:tokenUrl":"http://centralidp.tx.test/auth/realms/CX-Central/protocol/openid-connect/token",
      "oauth2:scope":"{{REGISTRY_TOKEN_SCOPE}}"
   }
}'
```

> 

Reviewer 2: ```suggestion
As per the current standards, there is no need to proxy the HTTP method. If there is (for instance due to a proprietary query endpoint), the data provider must ensure thatthat the technical user that the registered credentials abstract does NOT have the power to manipulate the DTR's data (for instance via the `PUT /shell-descriptors/{{aasId}}` API. 
```
